docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+
+A lot happens under the hood, though. All the user code now runs in unprivileged mode, and the systems services like the `NVIC` APIs or the OS SVCalls are routed through the uVisor.
+
+In the code above we specified 3 elements:
+
+1. System-wide uVisor configurations: `UVISOR_SET_PRIV_SYS_IRQ_HOOKS`. This macro is only needed temporarily, as it will be set by the operating system in the future.
+1. Main box ACLs. Since with uVisor enabled everything runs in unprivileged mode, we need to make sure that peripherals that are accessed by the OS and the main box are allowed. These peripherals are specified using a list like the one in the snippet above. The process is described in greater detail in a [section](#the-main-box-acls) below.
+1. App-specific uVisor configurations: `UVISOR_SET_MODE_ACL`. This macro sets the uVisor mode (enabled) and associates the list of ACLs we just created with the main box.
+
+Before compiling, we need to add a custom target that inherits all the features of the original `K64F` target, and enables the uVisor feature. To do so, add the file `~/code/uvisor-example/mbed_app.json` with the following content:
+
+```JSON
+{
+    "custom_targets": {
+        "K64F_SECURE": {
+            "inherits": ["K64F"],
+            "extra_labels_add":["K64F", "UVISOR_SUPPORTED"],
+            "features_add": ["UVISOR"]
+        }
+    }
+}
+```
+
+---
+
+**Checkpoint**
+
+Compile the application again, but this time targeting `K64F_SECURE`:
+
+```bash
+$ mbed compile -m K64F_SECURE -t GCC_ARM
+```
+
+The binary will be located at:
+
+```bash
+~/code/uvisor-example/.build/K64F_SECURE/GCC_ARM/uvisor-example.bin
+```
+
+Re-flash the device and press the reset button. The device LED should be blinking as in the previous case.
+
+## Add a secure box
+[Go to top](#overview)
+
+Now that uVisor is enabled, we can finally add a *secure box*.
+
+A secure box is a special compartment that is granted exclusive access to peripherals, memories and interrupts. Private resources are only accessible when the *context* of the secure box is active. The uVisor is the only one that can enable a secure box context, for example upon thread switching or interrupt handling.
+
+Code that belongs to a box is not obfuscated by uVisor, so it is still readable and executable from outside of the box. In addition, declaring an object in the same file that configures a secure box does not protect that object automatically.
+
+Instead, we provide specific APIs to instruct the uVisor to protect a private resource. Here we will show how to use these APIs in the `uvisor-example` app.
+
+### Configure the secure box
+
+For this example, we want to create a secure box called `private_timer`. The `private_timer` box has exclusive access to the PIT timer and to the GPIO PORT C on the NXP FRDM-K64F board. These peripherals cannot be accessed from outside the secure box.
```

Comment:
> These peripherals cannot be accessed from outside the secure box.

We intend to create a secure box that will prevent these peripherals from being accessed from outside the secure box.